### PR TITLE
Add Button component

### LINKED ISSUE #14: Button component
**Epic** https://github.com/x-team/auto/issues/573

---

Migrate https://github.com/x-team/auto/blob/master/src/client/components/Button.js and upgrade styles.

screen shot 2017-11-07 at 9 53 08 am

### PR DESCRIPTION
**Pull request by nicksp**
Closes #14 

Took a bit longer than planned initially, but mainly because I had to revise button styles and proposed props, and mainly re-structure it in a way that allows their combined usage, by applying new styles on top of base ones. As a result, we have a highly customisable component that allows to build it whatever we wish by combining various props :)

### Spec page

localhost_6060_

### CODE REVIEW

**Comment by joshwnj** [comment]
@nicksp is there a need for the `rounded` prop? It looks like all of the buttons in the new design have square corners.

**Comment by nicksp** [comment]
@joshwnj I was judging by new x-team.com website where I saw sever of rounded buttons. This prop is off by default and still might be a good addition to the component's possibilities.

**Comment by joshwnj** [comment]
Ok - let's leave it out for now, so that we're not adding complexity with no value in return.

File: src/components/Button.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      disabled,
+      rounded,
+      component: CustomComponent,
+      children,
+      ...rest
+    } = this.props
+
+    const colorClassName = colorStyles[color] || ''
+    const sizeClassName = sizeStyles[size] || ''
+    const extraClassName = cx({
```
Comment (joshwnj) [inline src/components/Button.js]: If we write it as:

```js
    const extraClassName = [
      rounded && extraStyles.rounded,
      outlined && extraStyles.outlined,
      outlined && 'outlined',
      disabled && extraStyles.disabled
    ].filter(Boolean).join(' ')
```

we don't need to add an extra package dependency.

Comment (nicksp) [reply src/components/Button.js]: sure, will adjust the code. I just thought that having `classnames` in place might be helpful for future work to simplify class creation logic ;) Also, getting rid of `rounded` prop.

Comment (joshwnj) [reply src/components/Button.js]: I'm fine with adding it, but only if we need it :) Once it's added, it's much more difficult to remove :)

**Comment by joshwnj** [comment]
Looks good 👍

**Pull request merged**